src/components/CreateArea.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import AddIcon from "@material-ui/icons/Add";
+import Fab from "@material-ui/core/Fab";
+import Zoom from "@material-ui/core/Zoom";
+
+function CreateArea(props) {
+  // useState returns two things in an array:
+  // the state and a function to setState
+  const [note, setNote] = useState({ title: "", content: "" });
+
+  //tracking the state of typing in the text content for UX rendering
+  const [typing, setTyping] = useState(false);
+
+  // function to display title field, will only be called if the user is typing a content
+  const showTitle = () => {
```

Comment:
That's pretty cool, I didn't know you can call a function that returns JSX inside of your JSX. Nice finding!